odl/contrib/README.md

```diff
@@ -0,0 +1,98 @@
+- To keep the history clean and to avoid accidental additions and deletions that bloat the repository, we will squash your pull request into a single commit that summarizes the whole addition. You can choose to do that yourself and write [a good commit message](https://odl.readthedocs.io/dev/gitwash/development_workflow.html#the-commit-message).
+- If no in-depth review was asked for, we will simply merge the PR if everything is okay. Otherwise, merging is delayed until you indicate that you have modified the code according to review.
+
+That's all.
+
+
+## Package layout
+
+All modules go into a subdirectory, e.g. `mylib`, in the `odl/contrib` directory. If you have code examples, put them into `example/contrib` (from the top level), again into a subdirectory of the same name as above, i.e. `mylib` here. The same applies for unit tests: put them into `odl/test/contrib`, making again a subdirectory of the same name as your module.
+
+In summary, the structure after adding your code would look like this:
+```
+ |
+ |---- odl
+ |      |
+ |      |-- contrib
+ |      |       |
+ |      ...     |-- __init__.py
+ |              |-- mylib
+ |              ...    |
+ |                     |-- __init__.py
+ |                     |-- mymodule1.py
+ |                     |-- mymodule2.py
+ |                     ...
+ |-- examples
+ |      |
+ |      |-- contrib
+ |      ...     |
+ |              |-- example1.py
+ |              |-- example2.py
+ |              ...
+ |
+ |-- tests
+ |      |
+ ...    |-- contrib
+        ...     |
+                |-- test_mymodule1.py
+                |-- test_mymodule2.py
+                ...
+```
+Your package will then be importable as `from odl.contrib import mylib`.
+
+If your code extends existing ODL functionality, consider using the same directory structure. For instance, if you add a new geometry, you may want to put it into `odl/contrib/my_new_geometry/tomo/geometry/mygeometry.py` and add a modified ray transform to `odl/contrib/my_new_geometry/tomo/operators/ray_trafo.py`.
+
+**Important:** The top-level `odl/contrib/__init__.py` should *never* be modified. In particular, it should not import any modules by default. This ensures that code in a submodule does not break or slow down the import of ODL as a whole.
+
+
+## Tests
+
+We recommend that you add [doctests](https://odlgroup.github.io/odl/dev/testing.html#doctests) to your code from the beginning. They are a natural way to show users examples of your code in action, while being like small unit tests. For complex functionality, you should consider adding [unit tests](https://odlgroup.github.io/odl/dev/testing.html#unit-tests).
+
+**Note:** If you add unit tests, they will be run together with the rest of the ODL test suite. This means that they must pass and should not take too long time.
```

Comment:
As mentioned somewhere, this should (imo) not be the case and these should only be run when then `--contrib` flag is used.